### PR TITLE
adding table functions for slicing, indexing, schema, column-names

### PR DESCRIPTION
**Pull request by vibhatha**

### CODE REVIEW

File: python/pycylon/data/table.pyx [hunk 1]
```diff
@@ -386,3 +386,14 @@ cdef class Table:
         py_arrow_table = pyarrow_wrap_table(table)
         return py_arrow_table.schema
 
+    def __getitem__(self, key):
+        table = to_pyarrow_table(self.id.encode())
+        py_arrow_table = pyarrow_wrap_table(table)
+        ctx = CylonContext(config=None)
```
Comment (nirandaperera) [inline python/pycylon/data/table.pyx:392]: shouldn't we take ctx from the table itself? this reinitialization would be problematic, isn't it?

Comment (vibhatha) [reply python/pycylon/data/table.pyx:392]: at the moment we don't support the wrapping and unwrapping C++ vs Python objects. We need to get that to do it. 
I am still exploring how to do that.

Comment (vibhatha) [reply python/pycylon/data/table.pyx:392]: There is an easy fix in Python class to get the context when it is being built. That is the straightforward way to do it.

Comment (vibhatha) [reply python/pycylon/data/table.pyx:392]: I checked this. Without Python CPP Object Wrapping Unwrapping, this won't be feasible at the moment.

Comment (vibhatha) [reply python/pycylon/data/table.pyx]: Added an issue. 
https://github.com/cylondata/cylon/issues/172
Need to work on this separately for a major refactor.

Comment (vibhatha) [reply python/pycylon/data/table.pyx]: @supunkamburugamuve @nirandaperera for this PR, the context creation issue was fixed.

**Review by nirandaperera — APPROVED** [review]

**Pull request merged**